core-api/src/test/java/com/optimizely/ab/event/internal/EventFactoryTest.java

```diff
@@ -315,6 +315,148 @@ public void createConversionEventIgnoresInvalidAndAcceptsValidAttributes() {
+        assumeTrue(datafileVersion >= Integer.parseInt(ProjectConfig.Version.V4.toString()));
+
+        Bucketer mockBucketAlgorithm = mock(Bucketer.class);
+        EventType eventType = validProjectConfig.getEventTypes().get(0);
+
+        List<Experiment> allExperiments = validProjectConfig.getExperiments();
+
+        // Bucket to the first variation for all experiments. However, only a subset of the experiments will actually
+        // call the bucket function.
+        for (Experiment experiment : allExperiments) {
+            when(mockBucketAlgorithm.bucket(experiment, userId))
+                .thenReturn(experiment.getVariations().get(0));
+        }
+        Attribute validFloatAttribute = validProjectConfig.getAttributes().get(0);
+        Attribute invalidFloatAttribute = validProjectConfig.getAttributes().get(1);
+        Attribute doubleAttribute = validProjectConfig.getAttributes().get(5);
+        Attribute integerAttribute = validProjectConfig.getAttributes().get(4);
+        Attribute boolAttribute = validProjectConfig.getAttributes().get(3);
+        Attribute emptyAttribute = validProjectConfig.getAttributes().get(6);
+
+        float validFloatValue = 2.1f;
+        float invalidFloatValue = (float) (Math.pow(2, 53) + 2000000000);
+        double invalidDoubleAttribute = Math.pow(2, 53) + 2;
+        long validLongAttribute = 12;
+        boolean validBoolAttribute = true;
+
+        Map<String, Object> eventTagMap = new HashMap<>();
+        eventTagMap.put("boolean_param", false);
+        eventTagMap.put("string_param", "123");
+
+        HashMap<String, Object> attributes = new HashMap<>();
+        attributes.put(validFloatAttribute.getKey(), validFloatValue);
+        attributes.put(invalidFloatAttribute.getKey(), invalidFloatValue);
+        attributes.put(doubleAttribute.getKey(), invalidDoubleAttribute);
+        attributes.put(integerAttribute.getKey(), validLongAttribute);
+        attributes.put(boolAttribute.getKey(), validBoolAttribute);
+        attributes.put(emptyAttribute.getKey(), validBoolAttribute);
+
+        LogEvent conversionEvent = factory.createConversionEvent(
+            validProjectConfig,
+            userId,
+            eventType.getId(),
+            eventType.getKey(),
+            attributes,
+            eventTagMap);
+
+        EventBatch impression = gson.fromJson(conversionEvent.getBody(), EventBatch.class);
+
+        //Check valid attributes are getting passed.
+        assertEquals(impression.getVisitors().get(0).getAttributes().get(0).getKey(), boolAttribute.getKey());
+        assertEquals(impression.getVisitors().get(0).getAttributes().get(0).getValue(), validBoolAttribute);
+        assertEquals(impression.getVisitors().get(0).getAttributes().get(1).getKey(), validFloatAttribute.getKey());
+        //In below condition I am checking Value of float with double value because impression gets visitors from json so that converts it into double
```

Comment:
nit. Change this to:
```
In the condition below we are checking Value of float with double value because impression gets visitors from JSON so that converts it into double
```